### DISCUSSION TITLE
How about  enable file filter for magit-diff  by default when  in  magit-log-mode buffer created by magit-log-buffer-file

### DISCUSSION THREAD

**Discussion by jixiuf** [magit/magit#4524] [Ideas]
When the major-mode of current buffer is `magit-log-mode` created by `magit-log-buffer-file`,
I hope the "Limit to files" can be `magit-current-file` automatically when magit-diff is called.
<img width="593" alt="image" src="https://user-images.githubusercontent.com/461743/136704272-38ab5d49-c711-4a6e-a87b-96ca5536ceff.png">

```
diff --git a/lisp/magit-diff.el b/lisp/magit-diff.el
index 88f2effb..71ca0c20 100644
--- a/lisp/magit-diff.el
+++ b/lisp/magit-diff.el
@@ -761,7 +761,7 @@ (defclass magit-diff-refresh-prefix (magit-diff-prefix)
                (magit-diff--get-value 'magit-diff-mode
                                       magit-prefix-use-buffer-arguments)))
     (unless (eq transient-current-command 'magit-dispatch)
-      (when-let ((file (magit-file-relative-name)))
+      (when-let ((file (magit-current-file)))
         (setq files (list file))))
     (oset obj value (if files `(("--" ,@files) ,args) args))))
 ```

**Comment by tarsius** [comment]
If you press `RET` with point on a the commit in a log, instead of `d c`, then that already is what happens.

Of course that means that you do not get a chance to change the other arguments up front and instead have to do, `RET` followed by e.g. `D - s g` instead of just `d - s c`. However I expect that you usually don't know before hand that the diff arguments that you have saved for use by default won't do the trick in this particular case, so you would usually show the diff with the configured arguments and then either be satisfied by the result or only then realize that in this particular case you need to use different arguments (such as a different value for `--diff-algorithm`).

**Reply by jixiuf** [reply]
> If you press `RET` with point on a the commit in a log, instead of `d c`, then that already is what happens.

But `d r` does not.

**Reply by tarsius** [reply]
Should it though? It might be nice to have the option to quickly get either behavior by using the respective binding.

**Reply by jixiuf** [reply]
It's ok for me by adding another binding or make `RET` showing the  differences between two commits when the region is active.

**Reply by tarsius** [reply]
That's a reasonable use-case. I'll look into it but it will likely take a while until I find the time.

**Reply by tarsius** [reply]
I don't remember when I implemented it, but this is how it works now.
